test-resources/diffs/5.patch

```diff
@@ -0,0 +1,4 @@
+diff --git a/src/formatting_stack/linters/line_length.clj b/src/formatting_stack/linters/line_length2.clj
+similarity index 100%
+rename from src/formatting_stack/linters/line_length.clj
```

Comment:
> The test-suite is adequate right now. 

> Having a test-suite testing every nook and cranny of git / patches is not a goal.

It's useful to remember the monorepo / git-status issues we had at some point; it took us a few iterations to get the solution right. Exhaustive testing+specing helped us gain confidence, once we had that in place we never saw an issue again.

More generally it's fairly uncontroversial that spec+tests precision correlate directly to bug-free software. They're one of the few automated tools we have directly at hand here and I'd appreciate their continued use just like every other feature in f-s.

This bug-freedom is particularly important for something that can be easily run as a blocking CI check.

Reply:
I'm not advocating against tests or questioning their usefulness in general.

We should be careful not to create numerous redundant test cases, they can hurt the clarity in case of an error or even a false sense of safety.

This function is not public api, and is only used by cljfmt, trivial-ns-duplicates, how-to-ns, clean-ns. Which is why the current test-suite is adequate.

Reply:
These aren't redundant tests because there's a very limited set of cases: 8 (1-7.patch to the proposed 8.patch). Each of them is _different_.

It's not like we're adding random tests that don't exercise any aspect in particular (we can call this _quantitative safety_). Instead, each test adds distinct coverage (qualitative safety).

When all known cases are covered (1-8), we get _exhaustive_ coverage. In this case we either have exhaustive coverage or not: it's a binary thing. Leaving the test suite "open" (non-exhaustive) opens the door to undetected regressions in a future; contrariwise a closed system is categorically superior.

A good example of exhaustiveness being deemed superior (because it rules out undefined behavior) can be found in the [exhaustive pattern matching](https://fsharpforfunandprofit.com/posts/correctness-exhaustive-pattern-matching/) construct, present in various FP langs.

Reply:
settled by https://github.com/nedap/formatting-stack/pull/154#issuecomment-750261251